mobs/nakki_v2/ai_states/NakkiEmergeOnTriggerState.cs

```diff
@@ -0,0 +1,93 @@
+using Godot;
+using System;
+using System.Linq;
+
+using Jakojaannos.WisperingWoods.Util.Editor;
+
+namespace Jakojaannos.WisperingWoods;
+
+[Tool]
+public partial class NakkiEmergeOnTriggerState : NakkiAiState {
+	public override string[] _GetConfigurationWarnings() {
+		return (base._GetConfigurationWarnings() ?? Array.Empty<string>())
+			.Union(this.CheckCommonConfigurationWarnings())
+			.ToArray();
+	}
+
+	[Export]
+	[MustSetInEditor]
+	public NakkiAiState? EnterStateAfterEmerge {
+		get => this.GetNotNullExportPropertyWithNullableBackingField(_enterStateAfterEmerge);
+		set => this.SetExportProperty(ref _enterStateAfterEmerge, value, notifyPropertyListChanged: true);
+	}
+	private NakkiAiState? _enterStateAfterEmerge;
+
+	[Export] private float _emergeAnimationSpeed = 3.0f;
+	[Export] private float _emergeDelay = 3.0f;
+
+	private Timer? _emergeOnTimeout;
+
+	private bool _emergeTimerDone = false;
+	private bool _initialDiveAnimationDone = false;
+	private bool _isEmerging = false;
+
+	public override void _Ready() {
+		if (Engine.IsEditorHint()) {
+			return;
+		}
+
+		_emergeOnTimeout = GetNode<Timer>("Timer");
```

Comment:
Would it make sense creating this child timer with `new Timer()` and `AddChild` instead of `GetNode`?

The timer `WaitTime` is managed here internally and it can't really be looping, so there isn't much that can/should be configured in the editor for the `Timer` node.

And if we didn't have the requirement to pre-wire these child timer nodes in the editor, there really isn't much reason to have e.g. the `nakki_emerge_on_trigger_state.tscn` scenes around, as those scenes would contain only the root-node.

And if we didn't have those state-type scenes around, we could mark the `NakkiSomeState` classes as `[GlobalClass]` so that they are available in editor via the "Add node" dialog. This could make the simpler state nodes a lot more ergonomic to use.

Reply:
It would be really nice to have these timers created on-demand with `GetTree().CreateTimer()`, but those timers don't have a `.Stop()` method, which complicates things.

For cases where the timer is only used to set a boolean, like this state class, it should be fine to just `_timer.TimeLeft = 0f`, though. This makes the timer finish immediately, which sadly sets the boolean, too, but this might not be that bad, mayb. _Maybe_ one-off timers could work. `AddChild` approach might still be cleaner, though :thinking: